jupyter-extension/jupyterlab_pachyderm/pfs_manager.py

```diff
@@ -3,11 +3,84 @@
 from jupyter_server.services.contents.filemanager import FileContentsManager
 import mimetypes
 from pachyderm_sdk import Client
-from pachyderm_sdk.api import pfs
+from pachyderm_sdk.api import pfs, pps
+import os
 from pathlib import Path
 from tornado import web
 
 
+def _create_base_model(path: str, fileinfo: pfs.FileInfo, type: str) -> dict:
+    if type == "file" and fileinfo.file_type != pfs.FileType.FILE:
+        raise web.HTTPError(400, f"{path} is not a file", reason="bad type")
+    if type == "directory" and fileinfo.file_type != pfs.FileType.DIR:
+        raise web.HTTPError(400, f"{path} is not a directory", reason="bad type")
+
+    model = {}
+    model["name"] = Path(path).name
+    model["path"] = path
+    model["created"] = fileinfo.committed
+    model["last_modified"] = fileinfo.committed
+    model["type"] = "directory" if fileinfo.file_type == pfs.FileType.DIR else "file"
+    model["content"] = None
+    model["mimetype"] = None
+    model["format"] = None
+    model["writable"] = False
+
+    return model
+
+
+def _get_file_model(
```

Comment:
I'm not a huge fan of methods that return nothing and mutate their inputs. Is there a reason as to why the `_get_file_model` and `_get_dir_model` methods don't call `_create_base_model` directly?